sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/DownloadManagerReceiver.java

```diff
@@ -33,7 +40,14 @@ public void onReceive(Context context, Intent intent) {
          */
         else if (DownloadManager.ACTION_DOWNLOAD_COMPLETE.equals(action)) {
             long downloadId = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, 0);
-            Distribute.getInstance().checkDownload(context, downloadId, false);
+
+            /* Check intent data is what we expected. */
+            long expectedDownloadId = SharedPreferencesManager.getLong(PREFERENCE_KEY_DOWNLOAD_ID, INVALID_DOWNLOAD_IDENTIFIER);
```

Comment:
Consider renaming to `DOWNLOAD_UNEXPECTED_IDENTIFIER`. Took some time to understand the logic here.

Reply:
Let's do NOT make more refactoring here. It isn't new code.